### PR TITLE
Satellite plotting fix, OMPS pairing fix, and some additional time interval looping

### PR DESCRIPTION
**Pull request by mbruckner-work**
Fixes to satplots from @blychs.
OMPS-NM level 3 ozone pairing fix necessary for calculating daily mean from model's xarray dataset. 
Enables time looped processing for RAQMS model, MOPITT CO obs. Small syntax change to allow use of time looping and OMPS-NM level 2 from Suomi NPP, NOAA-20, and future OMPS-NM instruments.

### CODE REVIEW

**Comment by blychs** [comment]
The code added by @mbruckner-work looks fine to me. Since this PR includes some of my own commits, I guess we want an independent reviewer though?
Could @rrbuchholz or @dwfncar do it?

**Review by dwfncar — APPROVED** [review]
All looks good.

File: melodies_monet/util/satellite_utilities.py
Comment (zmoon) [inline melodies_monet/util/satellite_utilities.py:144]: @blychs should there be `numeric_only=True` here?

Comment (zmoon) [reply melodies_monet/util/satellite_utilities.py:144]: And @mbruckner-work just curious why the 91ad0c3111b8b042e9c92efb359b42bec41e15e5 change? Shouldn't `.groupby('time.date').mean()` and `.resample(time='1D').mean()` give the same result?

Comment (mbruckner-work) [reply melodies_monet/util/satellite_utilities.py:144]: @zmoon The data is in xarray here. Putting in 'numeric_only=True' causes the code to fail.

Comment (blychs) [reply melodies_monet/util/satellite_utilities.py:144]: I believe that @mbruckner-work found that it didn't work, since it is a Dataset instead of a DataFrame, and needed to be taken out.

Comment (mbruckner-work) [reply melodies_monet/util/satellite_utilities.py:144]: The https://github.com/NOAA-CSL/MELODIES-MONET/commit/91ad0c3111b8b042e9c92efb359b42bec41e15e5 change was done because of the time format of the paired dataset. '.groupby('time.date').mean()' changes the time format to Timestamp while '.resample(time='1D').mean()' keeps the format as datetime64. 
The old way resulted in an error with the time filtering in the plotting code in my most recent tests, hence the change.

Comment (zmoon) [reply melodies_monet/util/satellite_utilities.py:144]: Thanks @mbruckner-work . Just wanted to make sure we are keeping up with @blychs efforts to supports pandas 2. You could remove the commented out lines that (no longer?) work.

**Comment by rschwant** [comment]
@mbruckner-work and @blychs, there is a small conflict now. Will Meng's method work here for you all too?

**Comment by blychs** [comment]
@rschwant It should work (and probably be a better solution than ours).

**Comment by rschwant** [comment]
I'm testing that Meng's update still works with the aircraft code now. Do you all want to test that it works for your satellite options? Or if you are pretty confident it will work. I can choose Meng's option. We can merge it in. And everyone can retest everything this afternoon.

**Comment by blychs** [comment]
I merged Meng's version, my code works fine and I don't think this can break Maggie's, but do you want to test it, just in case, @mbruckner-work ?

**Comment by mbruckner-work** [comment]
My test cases worked fine.

**Comment by blychs** [comment]
Then I think it could be merged, @rschwant , unless we need another review.

**Comment by rschwant** [comment]
I'm worried that both of your all's fix for the filter_criteria that Quazi set up breaks the altitude feature Quazi added @quaz115. I'm looking into it.

**Comment by blychs** [comment]
That's surprising. Is it breaking in the same line? Because we would expect both our previous solution and Meng's solution to not affect that, since the
```
if 'altitude' in ds_or_df:
```
should work worth for df and ds.
We could modify it to 
```
if 'altitude' in ds_or_df.keys():
```

or, worst case scenario, for a dirty solution
```
if isinstance(pairdf, pd.DataFrame) and 'altitude' in pairdf:
```

File: melodies_monet/util/satellite_utilities.py
Comment (zmoon) [reply melodies_monet/util/satellite_utilities.py:144]: It would be great to remove those lines before merging this.

Comment (zmoon) [reply melodies_monet/util/satellite_utilities.py:143]: Thanks @mbruckner-work appreciate it

**Comment by rschwant** [comment]
Okay it's fine. I'm having trouble calling the logic that Quazi set up as I don't fully understand the yaml options, but we update this later. The example runs well, so this can be pulled in.

**Pull request merged**
